File: perl/new_task.pl
#!/usr/bin/perl

use strict;
use warnings;

use Net::RabbitFoot;

my $conn = Net::RabbitFoot->new()->load_xml_spec()->connect(
    host => 'localhost',
    port => 5672,
    user => 'guest',
    pass => 'guest',
    vhost => '/',
    timeout => 1,
);


my $chan = $conn->open_channel();

$chan->declare_queue(
    queue => 'task_queue',
    durable => 1,
);

my $msg = join(' ', @ARGV) || "Hello World!";

$chan->publish(
    exchange => '',
    routing_key => 'task_queue',
    body => $msg,
);

print " [x] Sent '$msg'\n";

$conn->close();

File: dart/new_task.dart
import "package:dart_amqp/dart_amqp.dart";

void main(List<String> arguments) {
    ConnectionSettings settings = new ConnectionSettings(
        host: "localhost"
    );

    Client client = new Client(settings: settings);

    String consumeTag = "task_queue";
    String msg = arguments.isEmpty ? "Hello World!" : arguments[0];

    MessageProperties properties = new MessageProperties.persistentMessage();

    client
        .channel()
        .then((Channel channel) =>
                channel.queue(consumeTag, durable: true))
        .then((Queue queue) {
            queue.publish(msg, properties: properties);
            print(" [x] Sent ${msg}");
            return client.close();
        });
}
